File: SNI_UI2/CAPA_NEGOCIO/Questionnaires/Mapping/QuestionnairesDataBaseModel.cs
using CAPA_DATOS;
using CAPA_DATOS.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel
{
    public class Cat_Categorias_Test : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? id_categoria { get; set; }
        public string? descripcion { get; set; }
        public string? imagen { get; set; }
        public string? estado { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        [OneToMany(TableName = "Tests", KeyColumn = "id_categoria", ForeignKeyColumn = "id_categoria")]
        public List<Tests>? Tests { get; set; }

        internal object SaveCategoria()
        {
            imagen = FileService.setImage(imagen);
            return Save();
        }

        internal object UpdateCategoria()
        {
            imagen = FileService.setImage(imagen);
            return Update();
        }
    }
    public class Cat_Tipo_Preguntas : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? id_tipo_pregunta { get; set; }
        public string? tipo_pregunta { get; set; }
        public string? descripcion { get; set; }
        public string? estado { get; set; }
        public DateTime? fecha_crea { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public bool? editable { get; set; }
        public string? descripcion_general { get; set; }
        [OneToMany(TableName = "Cat_Valor_Preguntas", KeyColumn = "id_tipo_pregunta", ForeignKeyColumn = "id_tipo_pregunta")]
        public List<Cat_Valor_Preguntas>? Cat_Valor_Preguntas { get; set; }
        [OneToMany(TableName = "Pregunta_Tests", KeyColumn = "id_tipo_pregunta", ForeignKeyColumn = "id_tipo_pregunta")]
        public List<Pregunta_Tests>? Pregunta_Tests { get; set; }
    }
    public class Cat_Valor_Preguntas : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? id_valor_pregunta { get; set; }
        public string? descripcion { get; set; }
        public int? id_tipo_pregunta { get; set; }
        public string? estado { get; set; }
        public int? valor { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        [ManyToOne(TableName = "Cat_Tipo_Preguntas", KeyColumn = "id_tipo_pregunta", ForeignKeyColumn = "id_tipo_pregunta")]
        public Cat_Tipo_Preguntas? Cat_Tipo_Preguntas { get; set; }
        [OneToMany(TableName = "Resultados_Pregunta_Tests", KeyColumn = "id_valor_pregunta", ForeignKeyColumn = "id_valor_pregunta")]
        public List<Resultados_Pregunta_Tests>? Resultados_Pregunta_Tests { get; set; }
    }
    public class Pregunta_Tests : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? id_pregunta_test { get; set; }
        public string? estado { get; set; }
        public string? descripcion_pregunta { get; set; }
        public int? id_test { get; set; }
        public int? id_tipo_pregunta { get; set; }
        public DateTime? fecha { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public string? seccion { get; set; }
        public string? descripcion_general { get; set; }
        [ManyToOne(TableName = "Cat_Tipo_Preguntas", KeyColumn = "id_tipo_pregunta", ForeignKeyColumn = "id_tipo_pregunta")]
        public Cat_Tipo_Preguntas? Cat_Tipo_Preguntas { get; set; }
        [ManyToOne(TableName = "Tests", KeyColumn = "id_test", ForeignKeyColumn = "id_test")]
        public Tests? Tests { get; set; }
        [OneToMany(TableName = "Resultados_Pregunta_Tests", KeyColumn = "id_pregunta_test", ForeignKeyColumn = "id_pregunta_test")]
        public List<Resultados_Pregunta_Tests>? Resultados_Pregunta_Tests { get; set; }
    }
    public class Resultados_Pregunta_Tests : EntityClass
    {
        [PrimaryKey(Identity = false)]
        public int? Id_Perfil { get; set; }
        [PrimaryKey(Identity = false)]
        public int? id_pregunta_test { get; set; }
        public int? resultado { get; set; }
        public string? respuesta { get; set; }
        public string? estado { get; set; }
        public int? id_valor_pregunta { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        [PrimaryKey(Identity = false)]
        public DateTime? fecha { get; set; }
        [ManyToOne(TableName = "Cat_Valor_Preguntas", KeyColumn = "id_valor_pregunta", ForeignKeyColumn = "id_valor_pregunta")]
        public Cat_Valor_Preguntas? Cat_Valor_Preguntas { get; set; }
        [ManyToOne(TableName = "Pregunta_Tests", KeyColumn = "id_pregunta_test", ForeignKeyColumn = "id_pregunta_test")]
        public Pregunta_Tests? Pregunta_Tests { get; set; }
    }
    public class Resultados_Tests : EntityClass
    {
        [PrimaryKey(Identity = false)]
        public int? Id_Perfil { get; set; }
        [PrimaryKey(Identity = false)]
        public int? id_test { get; set; }
        [PrimaryKey(Identity = false)]
        public DateTime? fecha { get; set; }
        public string? seccion { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public string? valor { get; set; }
        public string? categoria_valor { get; set; }
        public string? tipo { get; set; }
        [ManyToOne(TableName = "Tests", KeyColumn = "id_test", ForeignKeyColumn = "id_test")]
        public Tests? Tests { get; set; }
    }
    public class Tests : EntityClass
    {
        [PrimaryKey(Identity = true)]
        public int? id_test { get; set; }
        public string? titulo { get; set; }
        public string? descripcion { get; set; }
        public int? grado { get; set; }
        public string? tipo_test { get; set; }
        public string? estado { get; set; }
        public int? id_categoria { get; set; }
        public DateTime? fecha_publicacion { get; set; }
        public DateTime? created_at { get; set; }
        public DateTime? updated_at { get; set; }
        public string? referencias { get; set; }
        public int? tiempo { get; set; }
        public int? caducidad { get; set; }
        public string? color { get; set; }
        public string? image { get; set; }
        [ManyToOne(TableName = "Cat_Categorias_Test", KeyColumn = "id_categoria", ForeignKeyColumn = "id_categoria")]
        public Cat_Categorias_Test? Cat_Categorias_Test { get; set; }
        [OneToMany(TableName = "Pregunta_Tests", KeyColumn = "id_test", ForeignKeyColumn = "id_test")]
        public List<Pregunta_Tests>? Pregunta_Tests { get; set; }
        //[OneToMany(TableName = "Resultados_Tests", KeyColumn = "id_test", ForeignKeyColumn = "id_test")]
        public List<Resultados_Tests>? Resultados_Tests { get; set; }

        internal object SaveTets()
        {
            image = FileService.setImage(image);
            return Save();
        }      

        internal object UpdateTest()
        {
            image = FileService.setImage(image);
            return Update();
        }
    }
}

File: SNI_UI2/wwwroot/Questionnaires/Model/QuestionnairesDataBaseModel.js
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../../WDevCore/WModules/WComponentsTools.js";
class Cat_Categorias_Test extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_categoria;
   /**@type {String}*/ descripcion;
   /**@type {String}*/ imagen;
   /**@type {String}*/ estado;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {Array<Tests>} OneToMany*/ Tests;
}
export { Cat_Categorias_Test }
class Cat_Tipo_Preguntas extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_tipo_pregunta;
   /**@type {String}*/ tipo_pregunta;
   /**@type {String}*/ descripcion;
   /**@type {String}*/ estado;
   /**@type {Date}*/ fecha_crea;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {Boolean}*/ editable;
   /**@type {String}*/ descripcion_general;
   /**@type {Array<Cat_Valor_Preguntas>} OneToMany*/ Cat_Valor_Preguntas;
   /**@type {Array<Pregunta_Tests>} OneToMany*/ Pregunta_Tests;
}
export { Cat_Tipo_Preguntas }
class Cat_Valor_Preguntas extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_valor_pregunta;
   /**@type {String}*/ descripcion;
   /**@type {String}*/ estado;
   /**@type {Number}*/ valor;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {Cat_Tipo_Preguntas} ManyToOne*/ Cat_Tipo_Preguntas;
   /**@type {Array<Resultados_Pregunta_Tests>} OneToMany*/ Resultados_Pregunta_Tests;
}
export { Cat_Valor_Preguntas }
class Pregunta_Tests extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_pregunta_test;
   /**@type {String}*/ estado;
   /**@type {String}*/ descripcion_pregunta;
   /**@type {Date}*/ fecha;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {String}*/ seccion;
   /**@type {String}*/ descripcion_general;
   /**@type {Cat_Tipo_Preguntas} ManyToOne*/ Cat_Tipo_Preguntas;
   /**@type {Tests} ManyToOne*/ Tests;
   /**@type {Array<Resultados_Pregunta_Tests>} OneToMany*/ Resultados_Pregunta_Tests;
}
export { Pregunta_Tests }
class Resultados_Pregunta_Tests extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Perfil;
   /**@type {Number}*/ resultado;
   /**@type {String}*/ respuesta;
   /**@type {String}*/ estado;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {Date}*/ fecha;
   /**@type {Cat_Valor_Preguntas} ManyToOne*/ Cat_Valor_Preguntas;
   /**@type {Pregunta_Tests} ManyToOne*/ Pregunta_Tests;
}
export { Resultados_Pregunta_Tests }
class Resultados_Tests extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Perfil;
   /**@type {Date}*/ fecha;
   /**@type {String}*/ seccion;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {String}*/ valor;
   /**@type {String}*/ categoria_valor;
   /**@type {String}*/ tipo;
   /**@type {Tests} ManyToOne*/ Tests;
}
export { Resultados_Tests }
class Tests extends EntityClass {
   constructor(props) {
       super(props, 'EntityQuestionnaires');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_test;
   /**@type {String}*/ titulo;
   /**@type {String}*/ descripcion;
   /**@type {Number}*/ grado;
   /**@type {String}*/ tipo_test;
   /**@type {String}*/ estado;
   /**@type {Date}*/ fecha_publicacion;
   /**@type {Date}*/ created_at;
   /**@type {Date}*/ updated_at;
   /**@type {String}*/ referencias;
   /**@type {Number}*/ tiempo;
   /**@type {Number}*/ caducidad;
   /**@type {String}*/ color;
   /**@type {String}*/ image;
   /**@type {Cat_Categorias_Test} ManyToOne*/ Cat_Categorias_Test;
   /**@type {Array<Pregunta_Tests>} OneToMany*/ Pregunta_Tests;
   /**@type {Array<Resultados_Tests>} OneToMany*/ Resultados_Tests;
}
export { Tests }
